boards/arm/mimxrt1064_evk/lpm.c

```diff
@@ -0,0 +1,303 @@
+{
+	/* Wait for flexspi to be inactive, and gate the clock */
+	while (!((FLEXSPI2->STS0 & FLEXSPI_STS0_ARBIDLE_MASK) &&
+			(FLEXSPI2->STS0 & FLEXSPI_STS0_SEQIDLE_MASK))) {
+	}
+	FLEXSPI2->MCR0 |= FLEXSPI_MCR0_MDIS_MASK;
+
+	/* Disable clock gate of flexspi2. */
+	CCM->CCGR7 &= (~CCM_CCGR7_CG1_MASK);
+}
+
+static void flexspi_exit_critical(void)
+{
+	/* Enable clock gate of flexspi2. */
+	CCM->CCGR7 |= (CCM_CCGR7_CG1_MASK);
+
+	FLEXSPI2->MCR0 &= ~FLEXSPI_MCR0_MDIS_MASK;
+	FLEXSPI2->MCR0 |= FLEXSPI_MCR0_SWRESET_MASK;
+	while (FLEXSPI2->MCR0 & FLEXSPI_MCR0_SWRESET_MASK) {
+	}
+	while (!((FLEXSPI2->STS0 & FLEXSPI_STS0_ARBIDLE_MASK) &&
+		(FLEXSPI2->STS0 & FLEXSPI_STS0_SEQIDLE_MASK))) {
+	}
+	/* Invalidate I-cache after flexspi clock changed. */
+	if (SCB_CCR_IC_Msk == (SCB_CCR_IC_Msk & SCB->CCR)) {
+		SCB_InvalidateICache();
+	}
+}
+
+void clock_full_power(void)
+{
+	/* PLL configs declared in soc folder */
+	extern clock_arm_pll_config_t armPllConfig;
+	extern clock_sys_pll_config_t sysPllConfig;
+	extern clock_usb_pll_config_t usb1PllConfig;
+	/* Power up PLLS */
+	/* Init ARM PLL */
+	CLOCK_SETDIV(kCLOCK_ArmDiv, 1);
+	CLOCK_InitArmPll(&armPllConfig);
+
+	/* Init SYS PLL*/
+	CLOCK_InitSysPll(&sysPllConfig);
+	/* Init System pfd0. */
+	CLOCK_InitSysPfd(kCLOCK_Pfd0, 24);
+	/* Init System pfd1. */
+	CLOCK_InitSysPfd(kCLOCK_Pfd1, 16);
+	/* Init System pfd2. */
+	CLOCK_InitSysPfd(kCLOCK_Pfd2, 29);
+	/* Init System pfd3. */
+	CLOCK_InitSysPfd(kCLOCK_Pfd3, 35);
+
+	/* Init USB2 PLL*/
+	CCM_ANALOG->PLL_USB2_SET = CCM_ANALOG_PLL_USB2_BYPASS_MASK;
+	CCM_ANALOG->PLL_USB2_SET = CCM_ANALOG_PLL_USB2_ENABLE_MASK;
+	CCM_ANALOG->PLL_USB2_SET = CCM_ANALOG_PLL_USB2_POWER_MASK;
+	while ((CCM_ANALOG->PLL_USB2 & CCM_ANALOG_PLL_USB2_LOCK_MASK) == 0) {
+	}
+	CCM_ANALOG->PLL_USB2_CLR = CCM_ANALOG_PLL_USB2_BYPASS_MASK;
+
+	/* Init AUDIO PLL */
+	CCM_ANALOG->PLL_AUDIO_SET = CCM_ANALOG_PLL_AUDIO_BYPASS_MASK;
+	CCM_ANALOG->PLL_AUDIO_CLR = CCM_ANALOG_PLL_AUDIO_POWERDOWN_MASK;
+	CCM_ANALOG->PLL_AUDIO_SET = CCM_ANALOG_PLL_AUDIO_ENABLE_MASK;
+	while ((CCM_ANALOG->PLL_AUDIO & CCM_ANALOG_PLL_AUDIO_LOCK_MASK) == 0) {
+	}
+	CCM_ANALOG->PLL_AUDIO_CLR = CCM_ANALOG_PLL_AUDIO_BYPASS_MASK;
+
+	/* Init VIDEO PLL */
+	CCM_ANALOG->PLL_VIDEO_SET = CCM_ANALOG_PLL_VIDEO_BYPASS_MASK;
+	CCM_ANALOG->PLL_VIDEO_CLR = CCM_ANALOG_PLL_VIDEO_POWERDOWN_MASK;
+	CCM_ANALOG->PLL_VIDEO_SET = CCM_ANALOG_PLL_VIDEO_ENABLE_MASK;
+	while ((CCM_ANALOG->PLL_VIDEO & CCM_ANALOG_PLL_VIDEO_LOCK_MASK) == 0) {
+	}
+	CCM_ANALOG->PLL_VIDEO_CLR = CCM_ANALOG_PLL_VIDEO_BYPASS_MASK;
+
+	/* Init ENET PLL */
+	CCM_ANALOG->PLL_ENET_SET = CCM_ANALOG_PLL_ENET_BYPASS_MASK;
+	CCM_ANALOG->PLL_ENET_CLR = CCM_ANALOG_PLL_ENET_POWERDOWN_MASK;
+	CCM_ANALOG->PLL_ENET_SET = CCM_ANALOG_PLL_ENET_ENABLE_MASK;
+	CCM_ANALOG->PLL_ENET_SET = CCM_ANALOG_PLL_ENET_ENET_25M_REF_EN_MASK;
+	while ((CCM_ANALOG->PLL_ENET & CCM_ANALOG_PLL_ENET_LOCK_MASK) == 0) {
+	}
+	CCM_ANALOG->PLL_ENET_CLR = CCM_ANALOG_PLL_ENET_BYPASS_MASK;
+
```

Comment:
We should probably check if the ETH, Video and Audio PLL's are used before enabling them. For e.g. `CONFIG_ETH_MCUX` could be used to check if ETH PLL should be enabled.

Reply:
I've updated the PR to do this in the latest push. I also added default PLL configurations in the `soc_rt10xx.c` file for the USB2 and AUDIO plls, based off the values chosen by the config tool

Reply:
Please do not use `INIT_USB_PLL `and `INIT_SYS_PLL `as these have been deleted. You could also avoid using `INIT_ARM_PLL`

Reply:
I've dropped `INIT_USB_PLL` and `INIT_SYS_PLL` in the latest push. I kept `INIT_ARM_PLL` for future compatibility with the RT1010-1024 parts, if we elect to enable this clock management code on those parts.